### PR TITLE
fix edit mode and add None to list

### LINKED ISSUE #1231: BUG- Two Data Node table views not refreshing
**Description**
When visualizing two tabular data nodes, the first one doesn't appear.

**How to reproduce**

Run this code.

```python

# Import necessary libraries
import pandas as pd
import taipy as tp
from taipy import Config, Scope
import datetime as dt


# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    return datetime, date, int_var, string_var


data = {"toto": [i for i in range(10_000)], 
        "titi": [2*i for i in range(10_000)],
        "tata": [4*i for i in range(10_000)]}



input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=data,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg]
)

scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg])

data_node = None

# GUI Markdown content
scenario_md = """
<|1 1|layout|
<|{scenario}|scenario_selector|>

<|{scenario}|scenario|>
|>

<|job_selector|>

<|1 1|layout|
Input 1 Dataset:
<|{scenario.input_1_data_node if scenario else None}|data_node|>

Input 2 Dataset:
<|{scenario.input_2_data_node if scenario else None}|data_node|>
|>


<|1 1|layout|
<|{data_node}|data_node_selector|>

<|{data_node}|data_node|>
|>
"""

# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)
    # Run the GUI
    tp.Gui(scenario_md).run(title="Bug replication", port=3248)
```

**Expected behavior**
We should be able to see both data nodes properly.

**Runtime environment**
 - Taipy: 3.1.1

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Create related issue in taipy-doc for documentation and Release Notes if relevant

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1231

### CODE REVIEW

File: frontend/taipy/src/DataNodeViewer.tsx [hunk 1]
```diff
@@ -1035,7 +1059,7 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
                                                                   format(dataValue as Date, "yyyy/MM/dd HH:mm:ss")
                                                                 : dtType == "float" && dtValue === null
                                                                 ? "NaN"
-                                                                : dtValue}
+                                                                : `${dtValue}`}
```
Comment (FabienLelaquais) [inline frontend/taipy/src/DataNodeViewer.tsx:1062]: Really??

File: frontend/taipy/src/ScenarioSelector.tsx [hunk 2]
```diff
@@ -548,15 +548,17 @@ const ScenarioSelector = (props: ScenarioSelectorProps) => {
...
-            <Tooltip title="Edit Scenario">
-                <IconButton
-                    data-id={props.id}
-                    onClick={openEditDialog}
-                    sx={tinyEditIconButtonSx}
-                    disabled={props.active}
-                >
-                    <EditOutlined />
-                </IconButton>
+            <Tooltip title={props.active ? "Edit Scenario" : "Can't edit Scenario"}>
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioSelector.tsx:551]: 👍

File: taipy/gui_core/_adapters.py [hunk 3]
```diff
@@ -172,11 +172,9 @@ def get_hash():
...
     @staticmethod
     def _is_tabular_data(datanode: DataNode, value: t.Any):
-        if isinstance(datanode, _TabularDataNodeMixin):
-            return True
-        if datanode.is_ready_for_reading:
-            return isinstance(value, (pd.DataFrame, pd.Series, list, tuple, dict))
-        return False
+        return isinstance(datanode, _TabularDataNodeMixin) or isinstance(
+            value, (pd.DataFrame, pd.Series, list, tuple, dict)
+        )
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py:177]: 🤞ready to be read

**Review by FabienLelaquais — APPROVED** [review]
😶

File: frontend/taipy/src/DataNodeViewer.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/DataNodeViewer.tsx:1062]: That's the consequence of the pseudo not ready ... Data not being recognized as tabular... Raised an error when trying to display! Wrong happen anymore

File: taipy/gui_core/_adapters.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_adapters.py:177]: I don't get why data is not ready so...

**Pull request merged**